sdk/inc/azure/iot/az_iot_adu_ota.h

```diff
@@ -0,0 +1,284 @@
+ *                                  instance with the results of the current or past
+ *                                  device update workflow, if available. Use NULL
+ *                                  if no results are available.
+ * @param[in] payload               An #az_span pointing to the memory buffer where to
+ *                                  write the resulting Azure Plug-and-Play properties.
+ * @param[out] out_payload          A pointer to hold the resulting #az_span slice of
+ *                                  `payload`, fitted to the content generated by this
+ *                                  function.
+ * @return An #az_result value indicating the result of the operation.
+ */
+AZ_NODISCARD az_result az_iot_adu_ota_get_properties_payload(
+    az_iot_hub_client const* iot_hub_client,
+    az_iot_adu_ota_device_information* device_information,
+    int32_t agent_state,
+    az_iot_adu_ota_workflow* workflow,
+    az_iot_adu_ota_install_result* last_install_result,
+    az_span payload,
+    az_span* out_payload);
+
+/**
+ * @brief Parses the json content from the ADU service writable properties into
+ *        a pre-defined structure.
+ *
+ * @param[in] iot_hub_client       A pointer to the #az_iot_hub_client instance.
+ * @param[in] jr                   A #az_json_reader initialized with the ADU
+ *                                 service writable properties json, set to the
+ *                                 beginning of the json object that is the value
+ *                                 of the ADU component.
+ * @param[in] buffer               An #az_span buffer where to write the parsed
+ *                                 values read from the json content.
+ * @param[out] update_request      A pointer to the #az_iot_adu_ota_update_request
+ *                                 structure where to store the parsed contents
+ *                                 read from the `jr` json reader.
+ *                                 In summary, this structure holds #az_span
+ *                                 instances that point to the actual data
+ *                                 parsed from `jr` and copied to `buffer`.
+ * @param[out] buffer_remainder    A pointer to an #az_span where to store the
+ *                                 remaining available space of `buffer`.
+ * @return An #az_result value indicating the result of the operation.
+ */
+AZ_NODISCARD az_result az_iot_adu_ota_parse_service_properties(
+    az_iot_hub_client const* iot_hub_client,
+    az_json_reader* jr,
+    az_span buffer,
+    az_iot_adu_ota_update_request* update_request,
+    az_span* buffer_remainder);
+
+/**
+ * @brief    Generates the payload necessary to respond to the service
+             after receiving incoming properties.
+ *
+ * @param[in] iot_hub_client    A pointer to the #az_iot_hub_client instance.
+ * @param[in] version           Version of the writable properties.
```

Comment:
Nit - need more clarification since version can refer to DU version (albeit hardcoded) or the $version thingy from PnP / device twin.  I think you mean the latter, right?  And of course "version" is very tied up into the version of the firmware itself, too.

So I'd go with not just a longer description but also like `properties_version` (or something better word-smithed) to make this really clear.